cmd/jaeger/internal/exporters/storageexporter/exporter_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storageexporter
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+)
+
+var (
+	storage_exporter *storageExporter
+	host             component.Host
+	storage_host     storageHost
+)
+
+type storageHost struct {
+	component *component.Component
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	myMap := make(map[component.ID]component.Component)
+	myMap[component.NewID("jaeger_storage")] = *host.component
```

Comment:
this is pretty misleading. jaeger-storage extension is not the same as storageexporter, but your test makes it look like they are the same.

Reply:
The storageexporter fetches the StorageFactory which should be of type `jaeger_storage` from the host ([ref](https://github.com/jaegertracing/jaeger/blob/main/cmd/jaeger/internal/exporters/storageexporter/exporter.go#L34)).
As a result, I configured the host in that way. Is there any better way to do this or should I explain it through comments in the code?

Reply:
ah, because your naming is confusing...